src/components/network-area-diagram-viewer/dynamic-css-utils.ts

```diff
@@ -9,77 +9,108 @@ export enum THRESHOLD_STATUS {
     BELOW = 'BELOW',
     ABOVE = 'ABOVE',
 }
+
+export enum CSS_RULE_TYPE {
+    THRESHOLD_DRIVEN = 'THRESHOLD_DRIVEN',
+    FUNCTION_DRIVEN = 'FUNCTION_DRIVEN ',
+}
+
 export type CSS_DECLARATION = Record<string, string>;
-export type CSS_RULE = {
+export type CSS_DECLARATION_CALLBACK = (value: number) => string;
+export type DYNAMIC_CSS_DECLARATION = Record<string, CSS_DECLARATION_CALLBACK | string>;
+
+type CSS_RULE_BASE = {
     cssSelector: string;
+    type: CSS_RULE_TYPE;
+};
```

Comment:
Code would be easier to read without this type.

Reply:
I removed and added a different way to check if it is function driven or threshold driven